src/Raven.Server/Smuggler/Documents/CsvStreamSource.cs

```diff
@@ -0,0 +1,213 @@
+            _collection = collection;
+        }      
+
+        public IDisposable Initialize(DatabaseSmugglerOptions options, SmugglerResult result, out long buildVersion)
+        {
+            buildVersion = 40;
+            _reader = new StreamReader(_stream);
+            _csvReader = new CsvReader(_reader);
+            _csvReader.Configuration.Delimiter = ",";            
+            return new DisposableAction(() =>
+            {
+                _reader.Dispose();
+                _csvReader.Dispose();
+            });
+        }
+
+        private bool _headersProcessed;
+
+        private void ProcessFieldsIfNeeded()
+        {
+            if(_headersProcessed)
+                return;            
+            for (var i = 0; i < _csvReader.FieldHeaders.Length; i++)
+            {
+                if (_csvReader.FieldHeaders[i].Equals(Constants.Documents.Metadata.Id))
+                {
+                    _hasId = true;
+                    _idIndex = i;
+                }
+                if (_csvReader.FieldHeaders[i].Equals(Constants.Documents.Metadata.Collection))
+                {
+                    _hasCollection = true;
+                    _collectionIndex = i;
+                }
+                if (_csvReader.FieldHeaders[i][0] == '@')
+                    continue;
+                var indexOfDot = _csvReader.FieldHeaders[i].IndexOf('.');
+                //We have a nested property
+                if (indexOfDot >= 0)
+                {
+                    if(_nestedPropertyDictionary == null)
+                        _nestedPropertyDictionary = new Dictionary<int, string[]>();
+                    _nestedPropertyDictionary[i] = _csvReader.FieldHeaders[i].Split('.', StringSplitOptions.RemoveEmptyEntries);
+                }
+            }
+            _headersProcessed = true;
+        }
+
+        public DatabaseItemType GetNextType()
+        {
+            var type = _currentType;
+            _currentType = DatabaseItemType.None;
+            return type;
+        }
+
+        public IEnumerable<DocumentItem> GetDocuments(List<string> collectionsToExport, INewDocumentActions actions)
+        {
+            while (_csvReader.Read())
+            {
+                ProcessFieldsIfNeeded();
+                yield return ConvertRecordToDocumentItem(_csvReader.CurrentRecord, _csvReader.FieldHeaders, _collection);
+            }
+            
+        }
+
+        private DocumentItem ConvertRecordToDocumentItem(string[] csvReaderCurrentRecord, string[] csvReaderFieldHeaders, string collection)
+        {
+            var idStr = _hasId ? csvReaderCurrentRecord[_idIndex] : _hasCollection? $"{csvReaderCurrentRecord[_collectionIndex]}/" : $"{collection}/";
+            var data = new DynamicJsonValue();
+            for (int i = 0; i < csvReaderFieldHeaders.Length; i++)
+            {
+                //ignoring reserved properties
+                if (csvReaderFieldHeaders[i][0] == '@')
```

Comment:
empty field name?

Reply:
That's not valid csv

Reply:
Like `Name,,Email` ? Note the double ,

Reply:
shouldn't happen in the headers line, can happen in the value lines